File: mcp/jcodemunch-mcp/tests/fixtures/go/sample.go
package sample

// MaxRetries is the retry limit.
const MaxRetries = 3

// User represents a user.
type User struct {
    ID   int
    Name string
}

// GetUser returns a user by ID.
func GetUser(id int) User {
    return User{ID: id}
}

// Authenticate checks a token.
func Authenticate(token string) bool {
    return len(token) > 0
}

File: mcp/jcodemunch-mcp/tests/fixtures/rust/sample.rs
const MAX_RETRIES: u32 = 3;

/// A user struct
struct User {
    id: u32,
    name: String,
}

impl User {
    /// Create a new user
    fn new(id: u32, name: String) -> Self {
        User { id, name }
    }
}

/// Authenticate a token
fn authenticate(token: &str) -> bool {
    !token.is_empty()
}
